packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -3704,119 +3467,547 @@ describe('TransactionController', () => {
+            status: TransactionStatus.confirmed,
+            time: 1,
+            txParams: { from: '0x1' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId2',
+            status: TransactionStatus.unapproved,
+            time: 2,
+            txParams: { from: '0x2' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId3',
+            status: TransactionStatus.submitted,
+            time: 1,
+            txParams: { from: '0x3' },
+          },
+        ];
+
+        expect(
+          controller.getTransactions({
+            searchCriteria: { time: 1 },
+            initialList: transactions,
+            filterToCurrentNetwork: false,
+          }),
+        ).toStrictEqual([transactions[0], transactions[2]]);
+      });
+
+      it('returns limited number of transactions sorted by descending time', () => {
+        const controller = newController();
+
+        const transactions: TransactionMeta[] = [
+          {
+            chainId: '0x1',
+            id: 'testId1',
+            status: TransactionStatus.confirmed,
+            time: 1,
+            txParams: { from: '0x1', nonce: '0x1' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId2',
+
+            status: TransactionStatus.unapproved,
+            time: 2,
+            txParams: { from: '0x2', nonce: '0x2' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId3',
+            status: TransactionStatus.submitted,
+            time: 3,
+            txParams: { from: '0x1', nonce: '0x3' },
+          },
+        ];
+
+        controller.state.transactions = transactions;
+
+        expect(
+          controller.getTransactions({
+            searchCriteria: { from: '0x1' },
+            filterToCurrentNetwork: false,
+            limit: 1,
+          }),
+        ).toStrictEqual([transactions[2]]);
+      });
+
+      it('returns limited number of transactions except for duplicate nonces', () => {
+        const controller = newController();
+
+        const transactions: TransactionMeta[] = [
+          {
+            chainId: '0x1',
+            id: 'testId1',
+            status: TransactionStatus.confirmed,
+            time: 1,
+            txParams: { from: '0x1', nonce: '0x1' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId2',
+
+            status: TransactionStatus.unapproved,
+            time: 2,
+            txParams: { from: '0x2', nonce: '0x2' },
+          },
+          {
+            chainId: '0x1',
+            id: 'testId3',
+            status: TransactionStatus.submitted,
+            time: 3,
+            txParams: { from: '0x1', nonce: '0x1' },
+          },
+        ];
+
+        controller.state.transactions = transactions;
+
+        expect(
+          controller.getTransactions({
+            searchCriteria: { from: '0x1' },
+            filterToCurrentNetwork: false,
+            limit: 1,
```

Comment:
Not a good test of limit, since there is only one nonce in the results anyways.

Reply:
Not sure I follow.

The search is still based on `from` meaning there would be two results, except the `limit` would then only return a single entry, except the duplicate nonce exception means we actually get 2 entries returned as we assert.

I've increased the limit anyway so we can also assert that we can have results with duplicate and unique nonces in.